### PR TITLE
[#621] Added docs for registration backends, and sectioned configuration docs

### LINKED ISSUE #621: Update documentation for registration backends
**Tasks**

- [x] Document (in English) how to configure registration backend: StUF-ZDS
- [x] Document (in English) how to configure registration backend: ZGW APIs
- [x] Document (in English) how to configure registration backend: Objects API
- [x] Document (in English) how to configure registration backend: Mail
- [x] Document (in English) how to configure registration backend: Empty (explain that they are then stored in OF)

See [here](https://open-forms.readthedocs.io/en/latest/configuration/bag.html) for examples

### PR DESCRIPTION
**Pull request by Bartvaderkin**
Fixes #621

### CODE REVIEW

File: docs/configuration/registration/objects.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+   a. Click **Service toevoegen**.
+   b. Fill out the form for each of the two components:
+
+      * **Label**: *For example:* ``Objecten`` or ``Documenten``
+      * **Type**: Select the type, one of: ``ORC`` or ``DRC``
+      * **API root url**: The root of this API, *for example* ``https://test.openzaak.nl/objecten/api/v1/``
+
+      * **Client ID**: Fill the value provided by the backend service *For example:* ``open-zaak`` (*NOTE* this could be different for each component)
+      * **Secret**: Fill the value provided by the backend service (*NOTE* this could be different for each component)
```
Comment (sergei-maertens) [inline docs/configuration/registration/objects.rst:24]: Objects API uses Token auth, so it's not the client ID + secret that are relevant, but rather the header (`Authentication`) and value: `Token <tokenValue>` that should be filled out, with the API key auth type.

DRC of course is client ID + secret

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: docs/configuration/registration/objects.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+1. In Open Forms, navigate to: **Configuration** > **Services**
+2. Here we need to configure credentials for two related API's:
+
+    - Objecten API (ORC)
+    - Documentregistratiecomponent (DRC)
+
+   a. Click **Service toevoegen**.
+   b. Fill out the form for each of the two components:
+
+      * **Label**: *For example:* ``Objecten`` or ``Documenten``
```
Comment (joeribekker) [inline docs/configuration/registration/objects.rst:19]: Can you split these in 2 separate steps? And explain that the submission of and attachments are stored in the documenten API

File: docs/configuration/registration/objects.rst [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+   c. Click **Opslaan** and repeat to create configuration for the other component.
+
+
+3. Navigate to **Configuration** > **Objecten API configuration**
+4. Enter the following details:
+
+   * **Objects API**: Select the Objecten API (ORC) service created above
+   * **Documenten API**: Select the Documentregistratiecomponent (DRC) service created above
+   * **Objecttype**: Fill in the default URL of the ProductAanvraag OBJECTTYPE in the Objecttypes API
+   * **Objecttype version**: Fill in the default version of the OBJECTTYPE in the Objecttypes API
```
Comment (joeribekker) [inline docs/configuration/registration/objects.rst:42]: Can you add example values for all fields below.

File: docs/configuration/registration/objects.rst [hunk 4]
```diff
@@ -0,0 +1,52 @@
+.. _configuration_registration_objects:
+
+============
+Objecten API
+============
+
+The Objecten API is a suite of REST based object registration services. Open Forms can be
+configured to access these API's to register form submissions.
+
```
Comment (joeribekker) [inline docs/configuration/registration/objects.rst:9]: @stevenbal could you add an example ProductAanvraag Objecttype here?

File: docs/configuration/registration/objects.rst [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    - Objecten API (ORC)
+    - Documentregistratiecomponent (DRC)
+
+   a. Click **Service toevoegen**.
+   b. Fill out the form for each of the two components:
+
+      * **Label**: *For example:* ``Objecten`` or ``Documenten``
+      * **Type**: Select the type, one of: ``ORC`` or ``DRC``
+      * **API root url**: The root of this API, *for example* ``https://test.openzaak.nl/objecten/api/v1/``
```
Comment (joeribekker) [inline docs/configuration/registration/objects.rst:21]: I think the use of OpenZaak domain for Objecten is confusing (altho it's a valid domain). Please a example.com domain.

**Review by joeribekker — CHANGES_REQUESTED** [review]

File: docs/configuration/registration/stufzds.rst [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+1. In Open Forms, navigate to: **Configuration** > **Soap Services**
+2. Click **Add Soap Services** and fill in the following details:
+
+   * **Label**: *A human readable label*, for example: `StUF-ZDS registration`
+
+3. In the **StUF parameters** section enter the sending and receiving organisation details provided by your backend provider
+4. In the **Connection** section:
+
+   * **SOAP Version**: select the SOAP version of your backend provider
+   * **URL**: fill in the base URL of the SOAP service
```
Comment (joeribekker) [inline docs/configuration/registration/stufzds.rst:19]: Nop, this value can be empty and the Beantwoordvraag, etc need to be full URLs

File: docs/configuration/registration/zgw.rst [hunk 7]
```diff
@@ -0,0 +1,58 @@
+.. _configuration_registration_zgw:
+
+=========
+ZGW API's
+=========
+
+The ZGW (Zaak Gericht Werken) API's are a suite of REST based Zaak services. Open Forms can be
```
Comment (joeribekker) [inline docs/configuration/registration/zgw.rst:7]:
```suggestion
The ZGW (Zaakgericht Werken) API's are a suite of REST based Zaak services. Open Forms can be
```

**Review by joeribekker — CHANGES_REQUESTED** [review]

File: docs/configuration/registration/objects.rst (hunk 4 above)
Comment (stevenbal) [reply docs/configuration/registration/objects.rst:9]: @joeribekker do you mean an example of a ProductAanvraag object, or the entire objecttype? This is what a ProductAanvraag could look like (`data` contains the actual form data the user entered)

```json
{
    "data": {
         "naam": "Jan Jansen",
         "omschrijving": "Ik heb een vraag over mijn paspoort",
         "telefoonnummer": "0612345678"
    },
    "type": "terugbelnotitie",
    "bsn": "111222333",
    "pdf_url": "https://openzaak.nl/documenten/api/v1/enkelvoudiginformatieobjecten/230bab4a-4b51-40c6-91b2-f2022008a7f8", 
    "attachments": [],
    "submission_id": "a43e84ac-e08b-4d5f-8d5c-5874c6dddf56"
}
```

Comment (Bartvaderkin) [reply docs/configuration/registration/objects.rst:9]: Thanks, I copied the whole structure and noted the top-level is meta-data of the submission and the data element has the form values.

**Review by sergei-maertens — APPROVED** [review]

**Review by joeribekker — APPROVED** [review]

**Pull request merged**